ingeniamc/ingenialink-python
# INGK-1012 Drive desconnection not detected when PDO poller is active

Fixes # INGK-1012

### Type of change

- Upon stopping the PDOs, only reset the slave states to PreOp if the slaves are connected and in either the Op, SafeOp, or SafeOp + Error states.

### Tests
Test 1:
- Connect to ML3.
- Run the PDO poller.
- Turn off the power supply.
- Check that the drive disconnection is detected.
- Turn on the power supply.
- Check that the drive re-connects.
- Run the PDO poller and check that it works.

Test 2:
- Connect to ML3
- Run the PDO poller.
- Disconnect the ethernet cable.
- Check that the drive disconnection is detected.
- Connect the ethernet cable.
- Check that the drive re-connects.
- Run the PDO poller and check that it works.

### Documentation


### Code formatting and linting

- [x] Use the ruff package to format the code: `ruff format ingenialink tests virtual_drive`.
- [x] Use the ruff package to lint the code: `ruff check ingenialink tests virtual_drive`.

### Others
